src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -75,30 +77,50 @@ private void init() throws Exception {
         currentProfile = sessionHelper.getFromMultipleProfiles();
     }
 
-    String run(Instant startInstant) throws Exception {
-        init();
+	String run(Instant startInstant) throws Exception {
+		init();
 
-        environment.awsRoleToAssume = currentProfile.map(profile1 -> profile1.roleArn).orElse(null);
+		environment.awsRoleToAssume = currentProfile.map(profile1 -> profile1.roleArn).orElse(null);
 
-        if (currentSession.isPresent() && sessionHelper.sessionIsActive(startInstant, currentSession.get()) &&
-                StringUtils.isBlank(environment.oktaProfile)) {
-            return currentSession.get().profileName;
-        }
+		if (currentSession.isPresent() && sessionHelper.sessionIsActive(startInstant, currentSession.get()) &&
+				StringUtils.isBlank(environment.oktaProfile)) {
+			return currentSession.get().profileName;
+		}
+
+		String samlResponse = oktaSaml.getSamlResponse();
+		AssumeRoleWithSAMLRequest assumeRequest = roleHelper.chooseAwsRoleToAssume(samlResponse);
+		Instant sessionExpiry = startInstant.plus(assumeRequest.getDurationSeconds() - 30, ChronoUnit.SECONDS);
+		AssumeRoleWithSAMLResult assumeResult = roleHelper.assumeChosenAwsRole(assumeRequest);
+		String profileName = profileHelper.createAwsProfile(assumeResult);
+
+		updateConfig(assumeRequest, sessionExpiry, profileName);
 
-        String samlResponse = oktaSaml.getSamlResponse();
-        AssumeRoleWithSAMLRequest assumeRequest = roleHelper.chooseAwsRoleToAssume(samlResponse);
-        Instant sessionExpiry = startInstant.plus(assumeRequest.getDurationSeconds() - 30, ChronoUnit.SECONDS);
-        AssumeRoleWithSAMLResult assumeResult = roleHelper.assumeChosenAwsRole(assumeRequest);
-        String profileName = profileHelper.createAwsProfile(assumeResult);
+		return profileName;
+	}
+    
+	AssumeRoleWithSAMLResult getAssumeRoleWithSAMLResult(Instant startInstant) throws Exception {
+		init();
 
-        environment.oktaProfile = profileName;
+		environment.awsRoleToAssume = currentProfile.map(profile1 -> profile1.roleArn).orElse(null);
+
+		String samlResponse = oktaSaml.getSamlResponse();
+		AssumeRoleWithSAMLRequest assumeRequest = roleHelper.chooseAwsRoleToAssume(samlResponse);
+		Instant sessionExpiry = startInstant.plus(assumeRequest.getDurationSeconds() - 30, ChronoUnit.SECONDS);
+		AssumeRoleWithSAMLResult assumeResult = roleHelper.assumeChosenAwsRole(assumeRequest);
+		String profileName = profileHelper.createAwsProfile(assumeResult);
```

Comment:
:art: Refactor this common code from here and **run(Instant)** into a shared method to keep this maintainable:

```java
String samlResponse = oktaSaml.getSamlResponse();
AssumeRoleWithSAMLRequest assumeRequest = roleHelper.chooseAwsRoleToAssume(samlResponse);
Instant sessionExpiry = startInstant.plus(assumeRequest.getDurationSeconds() - 30, ChronoUnit.SECONDS);
AssumeRoleWithSAMLResult assumeResult = roleHelper.assumeChosenAwsRole(assumeRequest);
String profileName = profileHelper.createAwsProfile(assumeResult);

updateConfig(assumeRequest, sessionExpiry, profileName);
```

If they are separate, we run the risk of fixing a bug for the interactive app and leaving it broken for the SDK.